src/pages/create/set/[id].tsx

```diff
@@ -20,8 +12,24 @@ export default function EditSet() {
   const [setDescription, setSetDescription] = useState("");
   const [flashcards, setFlashcards] = useState<TermDefPair[]>([]);
   const setId = useRouter().query.id as string;
+  const isEdit = !!(useRouter().query?.isEdit as string);
 
-  const { data: cards } = api.card.geCardBySet.useQuery(
+  const updateCard = ({
+    id,
+    term,
+    def,
+  }: {
+    id: number;
+    term: string;
+    def: string;
+  }) => {
+    if (flashcards?.[id]) {
+      flashcards[id].term = term;
```

Comment:
I think when we edit state objects, we want to make sure we use always use setState. So here is example syntax on how to do this:

In this example, `users` is the state and they iterate through all of the objects in `users` array and just basically create a new array while replacing the value they find at `idx`

```// Create a new array with the updated user
    const updatedUsers = users.map((user, idx) => {
      if (idx === index) {
        // This is the user we want to update
        return { ...user, first: newFirst, last: newLast };
      }
      return user; // This user doesn't need to be updated, return as is
    });